File: api/src/main/java/me/ramseyboy/exchange/api/domain/AreaCodeRepository.java
package me.ramseyboy.exchange.api.domain;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

public interface AreaCodeRepository extends JpaRepository<AreaCode, Integer> {
    Page<AreaCode> findByState(Pageable page, String state);

    Page<AreaCode> findByNpa(Pageable page, String npa);
}

File: api/src/main/kotlin/me/ramseyboy/exchange/api/domain/AreaCodeRepository.kt
package me.ramseyboy.exchange.api.domain

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface AreaCodeRepository : JpaRepository<AreaCode, Int> {
    fun findByState(page: Pageable, state: String): Page<AreaCode>

    fun findByNpa(page: Pageable, npa: String): Page<AreaCode>
}